### PR TITLE
Add Python 3.9 support

### LINKED ISSUE #1067: Failed import on python version 3.9.0
### Informations

- **Qiskit Aer version**: 0.7.1
- **Python version**: 3.9.0
- **Operating system**: Artix Linux

### What is the current behavior?
When importing qiskit in the python REPL, I get the following error message:

```python
>>> from qiskit import Aer
/home/ashish/.local/lib/python3.9/site-packages/qiskit/__init__.py:62: RuntimeWarning: Could not import the Aer provider from the qiskit-aer package. Install qiskit-aer or check your installation.
  warnings.warn('Could not import the Aer provider from the qiskit-aer '
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'Aer' from 'qiskit' (/home/ashish/.local/lib/python3.9/site-packages/qiskit/__init__.py)
```

However, doing a quick `pip list | grep qiskit` yields:

```
qiskit               0.23.1
qiskit-aer           0.7.1
qiskit-aqua          0.8.1
qiskit-ibmq-provider 0.11.1
qiskit-ignis         0.5.1
qiskit-terra         0.16.1
```

which means that `qiskit-aer` is installed, contrary to the error message.

### What is the expected behavior?
Importing Aer from qiskit must not yield error messages.

### Suggested solutions
One way is to downgrade to python 3.8.6 but I don't see why version 3.9.0 might have this issue.

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit adds official python 3.9 support to qiskit-aer. This
includes CI, release notes, and package metadata.

### Details and comments

Fixes #1067

Upstream dependencies that need Python 3.9 support:

~- [ ] osqp~ (sidestepped by skipping cvxpy tests for py39)

### CODE REVIEW

**Comment by mtreinish** [comment]
It looks like python 3.9 builds work fine (and passes verify wheels), but there are some CI issues for the tests jobs around cvxpy and related on python 3.9 that are blocking those

**Comment by mtreinish** [comment]
One thing to note here is that since #1074 we're not trying to test the noise transformations on Python 3.9 anymore as we're no longer installing cvxpy outside of python 3.7 and 3.8. There were issues installing cvxpy because of scs with 3.9.

**Review by vvilpas — CHANGES_REQUESTED** [review]
Looks good! Just a small typo.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -140,6 +140,8 @@ stages:
               python.version: '3.7'
             Python38:
               python.version: '3.8'
+            Python38:
```
[the remark this answers is not in the document]
Comment (mtreinish) [reply azure-pipelines.yml]: I fixed this issue yesterday to unblock the azure runs: https://github.com/Qiskit/qiskit-aer/pull/1071/commits/bb99ae1cb6283ae478c34799fe7a65aaaf7b1f61 I think you might have been looking at a stale diff.

[the remark this answers is not in the document]
Comment (vvilpas) [reply azure-pipelines.yml]: Right! Sorry about that.

**Review by vvilpas — APPROVED** [review]

**Pull request merged**